### ISSUE TITLE
Enum definition outside of the message

### ISSUE THREAD

**Issue by kubaj** [improbable-eng/ts-protoc-gen#14]
Enum definition doesn't work and protoc-gen-ts crashes:

```
protoc-gen-ts error: Error: No enum export for: ExampleType
    at /usr/lib/node_modules/ts-protoc-gen/lib/ts/message.js:85:23
    at Array.forEach (native)
    at Object.printMessage (/usr/lib/node_modules/ts-protoc-gen/lib/ts/message.js:38:38)
    at /usr/lib/node_modules/ts-protoc-gen/lib/ts/fileDescriptorTSD.js:29:33
    at Array.forEach (native)
    at Object.printFileDescriptorTSD (/usr/lib/node_modules/ts-protoc-gen/lib/ts/fileDescriptorTSD.js:28:41)
    at /usr/lib/node_modules/ts-protoc-gen/lib/ts_index.js:25:53
    at Array.forEach (native)
    at /usr/lib/node_modules/ts-protoc-gen/lib/ts_index.js:21:48
    at Socket.<anonymous> (/usr/lib/node_modules/ts-protoc-gen/lib/util.js:61:9)

--ts_out: protoc-gen-ts: Plugin failed with status code 1.
```
with this example:
```
enum ExampleType {
    TYPE_A = 0;
    TYPE_B = 1;
}

message ExampleMessage {
    ExampleType type = 1;
    string name = 2;
}
``` 

I've tried to add enum to the inside of the message:
```


message ExampleMessage {
    enum ExampleType {
        TYPE_A = 0;
        TYPE_B = 1;
    }
    ExampleType type = 1;
    string name = 2;
}
``` 
and compilations succeeds. Unfortunately this isn't solution for me because I need to add enum type to multiple messages.

**Comment by maxhodak** [comment]
+1 on this

**Comment by MarcusLongmuir** [comment]
Released in `0.3.3`

**Comment by TwelveNights** [comment]
I'm running into the same issue now. It looks like the behaviour persists in `ExportMap`, where `getEnum` is unaware of the scope.

```
protoc-gen-ts error: Error: No enum export for: MessageType
    at node_modules/ts-protoc-gen/lib/ts/message.js:86:23
    at Array.forEach (<anonymous>)
    at Object.printMessage (node_modules/ts-protoc-gen/lib/ts/message.js:38:38)
    at node_modules/ts-protoc-gen/lib/ts/fileDescriptorTSD.js:29:33
    at Array.forEach (<anonymous>)
    at Object.printFileDescriptorTSD (node_modules/ts-protoc-gen/lib/ts/fileDescriptorTSD.js:28:41)
    at node_modules/ts-protoc-gen/lib/ts_index.js:25:53
    at Array.forEach (<anonymous>)
    at node_modules/ts-protoc-gen/lib/ts_index.js:21:48
    at Socket.<anonymous> (node_modules/ts-protoc-gen/lib/util.js:61:9)
```
